pkg/util/azureclient/graphrbac_addons.go

```diff
@@ -0,0 +1,23 @@
+package azureclient
+
+import (
+	"context"
+	"fmt"
+)
+
+type ServicePrincipalsClientAddons interface {
+	ObjectIDforApplicationID(ctx context.Context, appID string) (string, error)
+}
+
+func (spc *servicePrincipalsClient) ObjectIDforApplicationID(ctx context.Context, appID string) (string, error) {
```

Comment:
1) reluctant to use addons for this - please can you put in a library function outside of pkg/util/azureclient
2) turns out this is also implemented in UpdateAADAppSecret() - please can you deduplicate?